src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineEdit.java

```diff
@@ -0,0 +1,165 @@
+    );
+    WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("confirmation-modal"));
+    WebElement changeSummaryEle = Helper.locateElementByCssSelector("textarea[data-testid=change-summary-input]");
+    ElementHelper.sendKeys(changeSummaryEle, testString);
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("Deploy"));
+    WaitHelper.waitForPageToLoad();
+    Assert.assertTrue(Helper.isElementExists(Helper.getCssSelectorByDataTestId("Deployed")));
+  }
+
+  @Then("Pipeline should show latest details")
+  public void pipelineShowLatestDetails() {
+    Assert.assertEquals(testString, getPipelineDetails("referenceName"));
+  }
+
+  @Then("History should show correct number of entries and info")
+  public void countHistoryVersions() {
+    openHistoryModalIfNotOpen();
+    List<WebElement> rows = Helper.locateElementsByXPath(pipelineHistoryRowXPath);
+    Assert.assertEquals(2, rows.size());
+    Assert.assertEquals(latestVersion, rows.get(0).findElement(By.cssSelector(
+      Helper.getCssSelectorByDataTestId("pipeline-history-date-label")
+    )).getText());
+    Assert.assertEquals(testString, rows.get(0).findElement(By.cssSelector(
+      Helper.getCssSelectorByDataTestId("pipeline-history-description")
+    )).getText());
+    Assert.assertEquals(olderVersion, rows.get(1).findElement(By.cssSelector(
+      Helper.getCssSelectorByDataTestId("pipeline-history-date-label")
+    )).getText());
+  }
+
+  @Then("View older version should show different details")
+  public void viewOlderVersion() {
+    List<WebElement> rows = Helper.locateElementsByXPath(pipelineHistoryRowXPath);
+    ElementHelper.clickOnElement(rows.get(1).findElement(By.cssSelector(
+      Helper.getCssSelectorByDataTestId("pipeline-history-view")
+    )));
+    WaitHelper.waitForPageToLoad();
+    Assert.assertFalse(testString.equals(getPipelineDetails("referenceName")));
+  }
+
+  @Then("Restore an older version should work")
+  public void restoreVersion() {
+    openHistoryModalIfNotOpen();
+    List<WebElement> rows = Helper.locateElementsByXPath(pipelineHistoryRowXPath);
+    ElementHelper.clickOnElement(rows.get(1).findElement(By.cssSelector(
+      Helper.getCssSelectorByDataTestId("pipeline-history-restore")
+    )));
+    Helper.waitSeconds(10);
+    Assert.assertFalse(testString.equals(getPipelineDetails("referenceName")));
+    openHistoryModalIfNotOpen();
+    rows = Helper.locateElementsByXPath(pipelineHistoryRowXPath);
+    Assert.assertEquals(3, rows.size());
+    Assert.assertTrue(rows.get(0).findElement(By.cssSelector(
+      Helper.getCssSelectorByDataTestId("pipeline-history-description")
+    )).getText().contains("Restore version at"));
+  }
+
+  public String getPipelineDetails(String dataTestId) {
+    Helper.waitSeconds();
+    CdfStudioActions.navigateToPluginPropertiesPage(fileSourceNode.getNodeName());
+    String value = Helper.locateElementByCssSelector("input[data-testid=" + dataTestId + "]").getAttribute("value");
```

Comment:
Similar to above, any reason not to use `Helper.locateElementByTestId` when looking up by `data-testid` value?